src/main/java/uk/gov/ons/census/fwmt/jobservice/nc/utils/DecryptNames.java

```diff
@@ -0,0 +1,118 @@
+package uk.gov.ons.census.fwmt.jobservice.nc.utils;
+
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.openpgp.PGPCompressedData;
+import org.bouncycastle.openpgp.PGPEncryptedDataList;
+import org.bouncycastle.openpgp.PGPException;
+import org.bouncycastle.openpgp.PGPLiteralData;
+import org.bouncycastle.openpgp.PGPObjectFactory;
+import org.bouncycastle.openpgp.PGPPrivateKey;
+import org.bouncycastle.openpgp.PGPPublicKeyEncryptedData;
+import org.bouncycastle.openpgp.PGPSecretKey;
+import org.bouncycastle.openpgp.PGPUtil;
+import org.bouncycastle.openpgp.jcajce.JcaPGPSecretKeyRing;
+import org.bouncycastle.openpgp.operator.jcajce.JcaKeyFingerprintCalculator;
+import org.bouncycastle.openpgp.operator.jcajce.JcePBESecretKeyDecryptorBuilder;
+import org.bouncycastle.openpgp.operator.jcajce.JcePublicKeyDataDecryptorFactoryBuilder;
+import org.bouncycastle.util.io.Streams;
+import uk.gov.ons.census.fwmt.common.error.GatewayException;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.Base64;
+import java.util.Iterator;
+
+public class DecryptNames {
+
+  public static String decryptFile(InputStream secretKeyFile, String householder, char[] passwd) throws GatewayException {
+    PGPPrivateKey secretKey;
+    PGPPublicKeyEncryptedData encryptedData = null;
+    Iterator<PGPPublicKeyEncryptedData> encryptedObjects;
+    long encryptedFileKeyId = 0;
+    try {
+      householder = householder.substring(1, householder.length() - 1);
+      encryptedObjects = getEncryptedObjects(Base64.getDecoder().decode(householder));
+    } catch (IOException e) {
+
+      throw new GatewayException(GatewayException.Fault.SYSTEM_ERROR, "Failed to obtain decryption data");
+    }
+    while (encryptedObjects.hasNext()) {
```

Comment:
do we need a loop here?  This is probably a decision point for the entire method , maybe better with an if statement - then nest the code below inside the if statement?